src/Core/Services/Implementations/CipherService.cs

```diff
@@ -790,5 +799,18 @@ private async Task<bool> UserCanEditAsync(Cipher cipher, Guid userId)
 
             return await _cipherRepository.GetCanEditByIdAsync(userId, cipher.Id);
         }
+
+        private async Task ValidateCipherLastKnownRevisionDateAsync(Cipher cipher, DateTime? lastKnownRevisionDate)
+        {
+            if (cipher.Id == default || !lastKnownRevisionDate.HasValue) return;
+
+            var existingCipher = await _cipherRepository.GetByIdAsync(cipher.Id);
+            if (existingCipher.RevisionDate != lastKnownRevisionDate)
```

Comment:
I am a little confused why you need to fetch from database. Can you just check RevisionDate on the passed in cipher parameter?

Reply:
The revision date on the incoming cipher is provided by the client, if validation is done against that, it should always pass.

That's why I (and Chad above) were wanting to validate based on the Cipher model's RevisionDate rather than adding a parameter

Reply:
The revision date is not provided by the client (or at least it should not). Nothing sets the revision date except the server just before save.

Reply:
I see what you're saying. All put methods grab the relevant ciphers from the data base, but the post methods don't validate that the cipher being given is a new one. It's definitely possible for a client to overwrite a cipher by supplying an id that exists.

This might be going afield of the issue at hand, though.